lib/url.py

```diff
@@ -14,11 +16,21 @@ def is_url_valid(url: str) -> bool:
     return True
 
 
-def is_url_valid_image(url: str) -> bool:
-    return is_url_valid(url) and is_url_image(url)
-
-
 # inspired by MattoTodd on StackOverflow https://stackoverflow.com/a/10543969
 def is_url_image(url: str) -> bool:
     mimetype = mimetypes.guess_type(url)
     return (mimetype and mimetype[0] and mimetype[0].startswith('image'))
+
+
+def is_url_valid_image(url: str) -> bool:
+    # check the returned returned content type
+    head = requests.head(url)
+    if head.headers.get('content-type').startswith('image'):
+        return True
+    elif head.status_code == 403:
+        # if HEAD isn't allowed, try a GET
+        get = requests.get(url)
+        if get.headers.get('content-type').startswith('image'):
+            return True
+
+    return False
```

Comment:
Nit: I think this is more readable
```suggestion
    if head.headers.get('content-type').startswith('image'):
        return True
    
    if head.status_code == 403:
        # if HEAD isn't allowed, try a GET
        get = requests.get(url)
        return get.headers.get('content-type').startswith('image')

    return False
```

Reply:
That's way cleaner, nice touch!